test/models/host_test.rb

```diff
@@ -678,45 +678,42 @@ def teardown
     refute_equal original_status, new_status
   end
 
-  test "assign a host to a location" do
-    host = Host.create :name => "host 1", :mac => "aabbecddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
-    location = Location.create :name => "New York"
-
-    host.location_id = location.id
-    assert host.save!
-  end
-
-  test "update a host's location" do
-    host = Host.create :name => "host 1", :mac => "aabbccddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
-    original_location = Location.create :name => "New York"
+  context 'host assigned to location and organization' do
+    setup do
+      @host = Host.create :name => "host 1", :mac => "aabbecddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
```

Comment:
Perhaps use FactoryBot to create the host? I don't think we care about any of these attributes, except possibly the `managed: false` one, and we might be getting some false positives - e.g. if the `hostgroups(:common)` has a location.

Reply:
Agree, also prefer `build` over `create` if you can afford it. I think this could be good fit, haven't seen all tests tho.